src/prefect/server/api/concurrency_limits.py

```diff
@@ -127,14 +298,53 @@ async def reset_concurrency_limit_by_tag(
     ),
     db: PrefectDBInterface = Depends(provide_database_interface),
 ) -> None:
+    # Reset V2 leases if V2 limit exists; otherwise fall back to V1
+    v2_name = f"tag:{tag}"
+
     async with db.session_context(begin_transaction=True) as session:
-        model = await models.concurrency_limits.reset_concurrency_limit_by_tag(
+        model = await cl_v2_models.read_concurrency_limit(session=session, name=v2_name)
+
+        if not model:
+            # Fall back to V1
+            model = await models.concurrency_limits.reset_concurrency_limit_by_tag(
+                session=session, tag=tag, slot_override=slot_override
+            )
+            if not model:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Concurrency limit not found",
+                )
+            return
+
+        # Revoke all existing leases for this limit
+        lease_storage = get_concurrency_lease_storage()
+        active_leases = await lease_storage.read_active_lease_ids(limit=1000)
+        for lease_id in active_leases:
+            lease = await lease_storage.read_lease(lease_id)
+            if lease and model.id in lease.resource_ids:
+                await lease_storage.revoke_lease(lease_id)
+
+        # Create new leases for slot_override if provided
+        if slot_override:
+            for task_run_id in slot_override:
+                holder = {"type": "task_run", "id": str(task_run_id)}
+                acquired = await cl_v2_models.bulk_increment_active_slots(
+                    session=session,
+                    concurrency_limit_ids=[model.id],
+                    slots=1,
+                )
+                if acquired:
+                    await lease_storage.create_lease(
+                        resource_ids=[model.id],
+                        ttl=V1_LEASE_TTL,
+                        metadata=ConcurrencyLimitLeaseMetadata(slots=1, holder=holder),
+                    )
+    # Mirror the reset into the V1 table so fallback reads stay accurate
+    async with db.session_context(begin_transaction=True) as session:
+        await models.concurrency_limits.reset_concurrency_limit_by_tag(
             session=session, tag=tag, slot_override=slot_override
         )
```

Comment:
The fallback read shouldn't be necessary since the V1 limit won't exist if the v2 limit exists.